rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/NotificationAndIndicationManager.java

```diff
@@ -64,122 +62,85 @@
         return Observable.defer(new Func0<Observable<Observable<byte[]>>>() {
             @Override
             public Observable<Observable<byte[]>> call() {
-                synchronized (NotificationAndIndicationManager.this) {
+                synchronized (activeNotificationObservableMap) {
                     final CharacteristicNotificationId id
                             = new CharacteristicNotificationId(characteristic.getUuid(), characteristic.getInstanceId());
 
-                    final HashMap<CharacteristicNotificationId, Observable<Observable<byte[]>>> conflictingServerInitiatedReadingMap
-                            = withAck ? notificationObservableMap : indicationObservableMap;
-                    final boolean conflictingNotificationIsAlreadySet
-                            = conflictingServerInitiatedReadingMap.containsKey(id);
-
-                    if (conflictingNotificationIsAlreadySet) {
-                        return Observable.error(new BleConflictingNotificationAlreadySetException(characteristic.getUuid(), !withAck));
-                    }
-
-                    final HashMap<CharacteristicNotificationId, Observable<Observable<byte[]>>> sameNotificationTypeMap
-                            = withAck ? indicationObservableMap : notificationObservableMap;
-
-                    final Observable<Observable<byte[]>> availableObservable = sameNotificationTypeMap.get(id);
+                    final ActiveCharacteristicNotification activeCharacteristicNotification = activeNotificationObservableMap.get(id);
 
-                    if (availableObservable != null) {
-                        return availableObservable;
+                    if (activeCharacteristicNotification != null) {
+                        if (activeCharacteristicNotification.isIndication == withAck) {
```

Comment:
`isIndication == withAck` is not clear for me. If I'd be a new to this domain I wouldn't know when it will be true.